test/unit/repositories/test_image.py

```diff
@@ -84,3 +86,84 @@ def test_create(self):
         self.mock_create(IMAGE_IN_DATA_ALL_VALUES)
         self.call_create()
         self.check_create_success()
+
+
+class ListDSL(ImageRepoDSL):
+    """Base class for `list` tests."""
+
+    _expected_image_out: list[ImageOut]
+    _entity_id_filter: Optional[str]
+    _primary_filter: Optional[bool]
+    _obtained_image_out: list[ImageOut]
+
+    def mock_list(self, image_in_data: list[dict]) -> None:
+        """
+        Mocks database methods appropriately to test the `list` repo method.
+
+        :param image_in_data: List of dictionaries containing the image data as would be required for an
+            `ImageIn` database model (i.e. no ID or created and modified times required).
+        """
+        self._expected_image_out = [
+            ImageOut(**ImageIn(**image_in_data).model_dump()) for image_in_data in image_in_data
+        ]
+
+        RepositoryTestHelpers.mock_find(
+            self.images_collection, [image_out.model_dump() for image_out in self._expected_image_out]
+        )
+
+    def call_list(self, entity_id: Optional[str] = None, primary: Optional[bool] = None) -> None:
+        """Calls the `ImageRepo` `list method` method.
+
+        :param entity_id: The ID of the entity to filter images by.
+        :param primary: The primary value to filter images by.
+        """
+        self._entity_id_filter = entity_id
+        self._primary_filter = primary
+        self._obtained_image_out = self.image_repository.list(
+            session=self.mock_session, entity_id=entity_id, primary=primary
+        )
+
+    def check_list_success(self) -> None:
+        """Checks that a prior call to `call_list` worked as expected."""
+        expected_query = {}
+        if self._entity_id_filter:
+            expected_query["entity_id"] = ObjectId(self._entity_id_filter)
+        if self._primary_filter != None:
```

Comment:
```suggestion
        if self._entity_id_filter is not None:
            expected_query["entity_id"] = ObjectId(self._entity_id_filter)
        if self._primary_filter is not None:
```
Resolves pylint issue
![image](https://github.com/user-attachments/assets/e0d5eef3-c946-46f2-8a52-2b2a6b16482d)